x-pack/plugins/infra/server/routes/inventory_metadata/index.ts

```diff
@@ -33,20 +34,35 @@ export const initInventoryMetaRoute = (libs: InfraBackendLibs) => {
     },
     async (requestContext, request, response) => {
       try {
-        const { sourceId, nodeType } = pipe(
+        const { sourceId, nodeType, currentTime } = pipe(
           InventoryMetaRequestRT.decode(request.body),
           fold(throwErrors(Boom.badRequest), identity)
         );
 
+        const model = findInventoryModel(nodeType);
+
+        // Only run this for AWS modules, eventually we might have more.
+        if (model.requiredModule !== 'aws') {
+          return response.ok({
+            body: {
+              accounts: [],
+              projects: [],
+              regions: [],
+            },
+          });
+        }
+
```

Comment:
Nice, I like bailing here. I think it might be better to do this check and bail here though:

https://github.com/elastic/kibana/blob/eafc72b8a758f83ef253c009b3e15137aaaa1870/x-pack/plugins/infra/server/routes/inventory_metadata/lib/get_cloud_metadata.ts#L32

where we already find the inventory model, and then we dictate the shape of the response in just one place and don't have to mimic it here. What do you think?

Reply:
I moved the check inside `getCloudMetadata()`